pages/_app.js

```diff
@@ -1,10 +1,28 @@
+import useLocalStorageState from "use-local-storage-state";
 import GlobalStyle from "../styles";
+import getCurrentTimeAndDate from "@/utils/getCurrentTimeAndDate";
+import { uid } from "uid";
 
 export default function App({ Component, pageProps }) {
+  const [tensionEntry, setTensionEntry] = useLocalStorageState("tensionEntry", {
+    defaultValue: [],
+  });
+
+  function handleAddTensionEntry(data) {
+    const timeStamp = getCurrentTimeAndDate();
+    const newEntry = { ...data, id: uid(), date: timeStamp };
+    setTensionEntry(
+      tensionEntry.length === 0 ? [newEntry] : [newEntry, ...tensionEntry]
+    );
```

Comment:
Not 100% sure , but the conditional seems redundant because spreading an empty array has the same effect as not spreading it at all. _Please proof this 😟_

```suggestion
    setTensionEntries([newEntry, ...tensionEntries]);
```